parsers.py

```diff
@@ -1,19 +1,113 @@
-from gcpy import read_geoschem_file
+import xarray as xr
+import numpy as np
+import yaml
+# from gcpy import read_geos_chem_file
 
-with open("config.yml", "r", encoding="utf8") as f:
+with open("config.yaml", "r", encoding="utf8") as f:
     config = yaml.safe_load(f)
 
 
-read_gc_file = read_geoschem_file # use gcpy function for reading GEOS-Chem files, may need to wrap
+def _open_geoschem(file_path, variables):
+    preprocess = lambda ds : ds[variables]
+    return xr.open_mfdataset(file_path, preprocess=preprocess)
 
 
-def read_TROPOMI_vXX(file_path):
+def read_geoschem_file(file_path_conc, file_path_edges):
+    '''
+    Eventually, this should be switched to a gcpy function. 
+    From Elise:
+        read_gc_file = read_geoschem_file 
+        # use gcpy function for reading GEOS-Chem files, may need to wrap
+    '''
+    # Define the variables that should be maintained when opening the files
+    fields = config["MODEL"]["DATA_FIELDS"]
+
+    conc_vars = dict(fields)
+    del conc_vars["PRESSURE_EDGES"]
+
+    edge_vars = dict(fields)
+    del edge_vars["CONC_AT_PRESSURE_CENTERS"]
+
+    # Open and combine edge and concentration files
+    gc = xr.merge([_open_geoschem(file_path_conc, list(conc_vars.values())),
+                   _open_geoschem(file_path_edges, list(edge_vars.values()))])
+
+    # Rename the fields to the standard (as defined in config.yaml)
+    rename_fields = {v : k for k, v in fields.items()}
+    gc = gc.rename(rename_fields)
+
+    # Transpose
+    gc = gc.transpose("TIME", "LONGITUDE", "LATITUDE", "LEV", "ILEV")
+
+    #  TO DO: Check if we need to fill the first hour of the data.
+
+    return gc
+
+
+def read_satellite_file(file_path, satellite_name):
```

Comment:
hmm wondering if this should be a class you inherit + modify... or something...

Reply:
I am not sure if we should have a generic function, as I think very few cases it would work... so we should encourage people to write a new one if their instrument isn't supported